pkg/cliutil/prepare/prepare.go

```diff
@@ -36,36 +35,48 @@ var (
 	errDeployPortConflict = errNSDeploy.NewType("port_conflict", errutil.ErrTraitPreCheck)
 )
 
-func fixDir(topo *spec.Specification) func(string) string {
+func fixDir(topo spec.Topology) func(string) string {
 	return func(dir string) string {
 		if dir != "" {
-			return clusterutil.Abs(topo.GlobalOptions.User, dir)
+			return clusterutil.Abs(topo.BaseTopo().GlobalOptions.User, dir)
 		}
 		return dir
 	}
 }
 
 // CheckClusterDirConflict checks cluster dir conflict
-func CheckClusterDirConflict(clusterSpec *meta.SpecManager, clusterName string, topo *spec.Specification) error {
+func CheckClusterDirConflict(clusterSpec *spec.SpecManager, clusterName string, topo spec.Topology) error {
 	type DirAccessor struct {
 		dirKind  string
-		accessor func(spec.Instance, *spec.Specification) string
+		accessor func(spec.Instance, spec.Topology) string
 	}
 
 	instanceDirAccessor := []DirAccessor{
-		{dirKind: "deploy directory", accessor: func(instance spec.Instance, topo *spec.Specification) string { return instance.DeployDir() }},
-		{dirKind: "data directory", accessor: func(instance spec.Instance, topo *spec.Specification) string { return instance.DataDir() }},
-		{dirKind: "log directory", accessor: func(instance spec.Instance, topo *spec.Specification) string { return instance.LogDir() }},
+		{dirKind: "deploy directory", accessor: func(instance spec.Instance, topo spec.Topology) string { return instance.DeployDir() }},
+		{dirKind: "data directory", accessor: func(instance spec.Instance, topo spec.Topology) string { return instance.DataDir() }},
+		{dirKind: "log directory", accessor: func(instance spec.Instance, topo spec.Topology) string { return instance.LogDir() }},
 	}
 	hostDirAccessor := []DirAccessor{
-		{dirKind: "monitor deploy directory", accessor: func(instance spec.Instance, topo *spec.Specification) string {
-			return topo.MonitoredOptions.DeployDir
+		{dirKind: "monitor deploy directory", accessor: func(instance spec.Instance, topo spec.Topology) string {
+			m := topo.BaseTopo().MonitoredOptions
```

Comment:
Monitor options shouldn't be missing.

Reply:
This `MonitoredOptions` control thinks about node/backbox export, I don't think this is a *must* for deploying a cluster although tiup-cluster always having this now.